tests/dragonfly/instance.py

```diff
@@ -426,8 +426,7 @@ def create(self, existing_port=None, path=None, version=100, **kwargs) -> DflyIn
         args.setdefault("log_dir", self.params.log_dir)
 
         if version >= 1.21 and "serialization_max_chunk_size" not in args:
-            # Add 1 byte limit for big values
-            args.setdefault("serialization_max_chunk_size", 1)
+            args.setdefault("serialization_max_chunk_size", 16384)
```

Comment:
if the huge value in seeder is 512, how come serialization_max_chunk_size=16384 will get to check the big value serialisation flow? I am missing something in the math.
lets say we have a backet that contains this key with big value elements, the number of big value entry is 2 so the this big value key serialization size is ~1K so how setting serialization_max_chunk_size=16384 makes us preempt in this container serialization?

Reply:
It's simple. We flush every `32k` (look PushSerializedToChannel in IterateBucketsFb), a bucket has 12 slots on 512 bytes each (and big value is 1k) so:

First bucket -> 6k serialized (no big values) and no preemptions
Second bucket -> 12k serialized(no big values) and no preemptions
Third bucket -> slot 1, slot 2... slot 7 = 15.5k serialized -> next `slot` overshoots and gets Flushed 
Fourth bucket -> 0k serialized start again

So we hit it every third bucket. IMO finding a good value for this is hard, the important part is as you said to hit both flows. Furthermore, if you increase the load and add some really big values we will fail the test since the gh runner is only 8gigs and we have a test case with 1 million elements which roughlty gives us 10k big values

Reply:
I understand. so in this case the huge values that you insert actually does not trigger this preemption but instead its just the fact that we flush only if the buffer is >32k, right?
so the preemptions stats are not really related to the number of big values inserted by seeder (because they are not really big) but instead they are related to the average bucket serialze size.
so if the bucket serialized size is 6k you will preempt every 3rd bucket.
which actually means that while the max chuck size is lower than 32K the big value serialziation logic will be triggered a lot a not only if we have big values in this bucket.

Reply:
> I understand. so in this case the huge values that you insert actually does not trigger this preemption but instead its

It might or might not. Depend on which value "breaks" the threshold

Reply:
in our server the value of serialization_max_chunk_size should be bigger than 32K the constant min blob size.
for tests just to make sure our logic works well its ok to make serialization_max_chunk_size smaller, but I actually dont see the point of the seeder with big values in this flow, as they are not realy big to trigger this flow

Reply:
> but I actually dont see the point of the seeder with big values in this flow, as they are not realy big to trigger this flow

Yep and for that we can add a specific test case which can also be fast to run (with relatively low load but huge values). That way we get both :)

Reply:
so if you agree with me on that lets make the default huge_value_percentage = 0 and huge_value_size = 10000
and we can just add a specific test case where huge_value_percentage = 2